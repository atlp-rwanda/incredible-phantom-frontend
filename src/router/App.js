import React from 'react';
import { LandingPage } from '../pages/LandingPage/LandingPage';
import { About } from '../pages/About/About';
import { NotFound } from '../components/NotFound/NotFound';
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';
import { Contact } from '../pages/Contact/Contact';
import { Helmet } from 'react-helmet';
import { useTranslation } from 'react-i18next';
import Forgot from '../components/resetPassword/Forgot';
import Reset from '../components/resetPassword/Reset';
import SigninPage from '../pages/SigninPage';

const App = () => {
  const { t } = useTranslation();
  return (
    <Router>
      <Helmet>
        <title>{t('webtitle')}</title>
      </Helmet>
      <Switch>
        <Route path='/' exact component={LandingPage} />
        <Route path='/about' exact component={About} />
        <Route path='/contact' exact component={Contact} />
        <Route path='/forgotpassword' exact component={Forgot} />
        <Route path='/reset' exact component={Reset} />
        <Route path='/signin' component={SigninPage} />
        <Route component={NotFound} />
      </Switch>
    </Router>
  );
};
export default App;
