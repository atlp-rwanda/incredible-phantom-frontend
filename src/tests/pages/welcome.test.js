import React from 'react';
import { shallow } from 'enzyme';
import App from '../../router/App';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

describe('home page', () => {
  it('should return the welcoming page', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
