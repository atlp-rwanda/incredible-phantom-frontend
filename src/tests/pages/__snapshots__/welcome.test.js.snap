// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page should return the welcoming page 1`] = `
<BrowserRouter>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      webtitle
    </title>
  </HelmetWrapper>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/about"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/contact"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
